src/order/basic.lean

```diff
@@ -198,6 +201,19 @@ protected lemma decidable.ne_iff_lt_iff_le [partial_order α] [@decidable_rel α
 @[simp] lemma ne_iff_lt_iff_le [partial_order α] {a b : α} : (a ≠ b ↔ a < b) ↔ a ≤ b :=
 by haveI := classical.dec; exact decidable.ne_iff_lt_iff_le
 
+lemma lt_iff_ne_and_le [partial_order α] (a b : α) : a < b ↔ (a ≠ b) ∧ a ≤ b :=
```

Comment:
Note that `lt_iff_le_and_ne` exists. Do you really need this? (If so, please simplify the proof.)

Reply:
No I don't. Thanks.